.github/workflows/misc.yml

```diff
@@ -79,11 +83,24 @@ jobs:
           export GOTEST_FLAGS="-v -p 1 -timeout=30m -coverprofile=coverage.out -covermode=atomic"
           make ${{ matrix.args }}
 
-      - if: runner.os == 'Linux'
+      - uses: actions/upload-artifact@v4
+        if: ${{ runner.os == 'Linux' && matrix.goversion == '1.22.x' }}
```

Comment:
Just curious, why is the go version important?

Reply:
a-ha, this one actually doesn't need it, because goversion is always 1.22

For the other cases, it is needed to avoid the workflow running 2 times. I'd say to keep the conditional here even if it is a no-op, so we keep this section the same as the one in all other files